dtfs-central-api/src/v1/controllers/utilisation-report-service/get-selected-fee-records-details.controller/helpers.test.ts

```diff
@@ -0,0 +1,125 @@
+
+      // Assert
+      expect(result.feeRecords[0].reportedPayment).toEqual<CurrencyAndAmount>({ amount: 200, currency: 'USD' });
+    });
+
+    it('sets the reported payment to the fees paid to ukef for the period converted into the payment currency when the payment currency does not match the fees paid to ukef for the period currency', async () => {
+      // Arrange
+      const feeRecord = FeeRecordEntityMockBuilder.forReport(UtilisationReportEntityMockBuilder.forStatus('PENDING_RECONCILIATION').build())
+        .withFeesPaidToUkefForThePeriod(2200)
+        .withFeesPaidToUkefForThePeriodCurrency('EUR')
+        .withPaymentCurrency('GBP')
+        .withPaymentExchangeRate(1.1)
+        .build();
+
+      // Act
+      const result = await mapToSelectedFeeRecordDetails('123', aReportPeriod(), [feeRecord]);
+
+      // Assert
+      expect(result.feeRecords[0].reportedPayment).toEqual<CurrencyAndAmount>({ amount: 2000, currency: 'GBP' });
+    });
+
+    it('maps the fees paid to ukef for the period to the reported fee', async () => {
+      // Arrange
+      const feeRecord = FeeRecordEntityMockBuilder.forReport(UtilisationReportEntityMockBuilder.forStatus('PENDING_RECONCILIATION').build())
+        .withFeesPaidToUkefForThePeriod(2200)
+        .withFeesPaidToUkefForThePeriodCurrency('EUR')
+        .build();
+      // Act
+      const result = await mapToSelectedFeeRecordDetails('123', aReportPeriod(), [feeRecord]);
+
+      // Assert
+      expect(result.feeRecords[0].reportedFee).toEqual<CurrencyAndAmount>({ amount: 2200, currency: 'EUR' });
+    });
+
+    it('calculates total reported payments in payment currency by adding up reported payments of all fee records', async () => {
+      // Arrange
+      const feeRecordWithDifferingCurrencies = FeeRecordEntityMockBuilder.forReport(
+        UtilisationReportEntityMockBuilder.forStatus('PENDING_RECONCILIATION').build(),
+      )
+        .withFeesPaidToUkefForThePeriod(2200)
+        .withFeesPaidToUkefForThePeriodCurrency('EUR')
+        .withPaymentCurrency('GBP')
+        .withPaymentExchangeRate(1.1)
+        .build();
+
+      const feeRecordWithMatchingCurrencies = FeeRecordEntityMockBuilder.forReport(
+        UtilisationReportEntityMockBuilder.forStatus('PENDING_RECONCILIATION').build(),
+      )
+        .withFeesPaidToUkefForThePeriod(1000)
+        .withFeesPaidToUkefForThePeriodCurrency('GBP')
+        .withPaymentCurrency('GBP')
+        .build();
+
+      // Act
+      const result = await mapToSelectedFeeRecordDetails('123', aReportPeriod(), [feeRecordWithDifferingCurrencies, feeRecordWithMatchingCurrencies]);
+
+      // Assert
+      expect(result.totalReportedPayments).toEqual<CurrencyAndAmount>({ amount: 3000, currency: 'GBP' });
+    });
+  });
+
+  function aFeeRecord() {
+    return FeeRecordEntityMockBuilder.forReport(UtilisationReportEntityMockBuilder.forStatus('RECONCILIATION_IN_PROGRESS').build()).build();
+  }
```

Comment:
So I wrote this as a function over an arrow function so I could write it at the bottom, and I want it at the bottom because the idea is it is fairly irrelevant what it does, so it's nice to not have it cluttering the top of the file.